src/components/views/FootnotesSelector.tsx

```diff
@@ -111,7 +111,10 @@ export const FootnotesSelector: React.FC<FootnotesSelectorProps> = ({
   }
 
   const handleClick = () => {
-    return onInsert(footnotes.filter((node) => selections.has(node.attrs.id)))
+    const selectedFootnotes = footnotes.filter((node) =>
+      selections.has(node.attrs.id)
+    )
+    onInsert(selectedFootnotes)
```

Comment:
why did we need to change this?

Reply:
This change ensures that the onInsert function is always called, even when there are no selected footnotes. In the previous code, handleClick only returned the filtered list of footnotes but didn’t trigger any further action if the list was empty. This became an issue specifically when unciting the last selected inline footnote, as selectedFootnotes would be empty, and onInsert wasn’t executed.

By calling onInsert with the filtered results (even if selectedFootnotes is zero), we can  handle the case of deleting the inline footnote

Reply:
The logic looks identical to me, onInsert is always called in the old code as well.